.github/workflows/pr.yaml

```diff
@@ -0,0 +1,32 @@
+name: Pull Request
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths: ['**/*.json', '**/*.jsonc', '**/*.ts']
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+      - uses: pnpm/action-setup@v4
+        with:
+          run_install: false
```

Comment:
we need to install pnpm before installing deps.
We want to skip installing the deps at this step because we want to restore the cache first.

https://github.com/pnpm/action-setup?tab=readme-ov-file#use-cache-to-reduce-installation-time